app/models/generic_object_definition.rb

```diff
@@ -7,6 +16,18 @@ class GenericObjectDefinition < ApplicationRecord
   has_many  :generic_objects
 
   def defined_attributes
-    @defined_attributes ||= properties[:attributes].stringify_keys
+    properties[:attributes]
+  end
+
+  def properties=(props)
+    props = props.symbolize_keys
+    if props.key?(:attributes)
+      # unify attribute name as string and attribute type as symbol
```

Comment:
unify is a strange word to use here...perhaps just remove the comment altogether, as it seems pretty straighforward.